pkg/app/ops/insightcollector/mode.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightcollector
+
+import (
+	"bytes"
+)
+
+type CollectorMode uint
+
+// Options controlling the InsightCollector.
+// The value is a sequence of zero or more of these letters:
+// C: enable data collection for [C]hange failure rate .
+// D: enable data collection for [D]evelopment frequency.
+// A: enable data collection for [A]pplication count.
+const (
+	EnableChangeFailureRate CollectorMode = 1 << iota
+	EnableDevelopmentFrequency
+	EnableApplicationCount
+)
+
+func NewCollectorMode() CollectorMode {
+	return CollectorMode(0)
+}
+
+func (m CollectorMode) String() string {
+	var buf bytes.Buffer
+	if m.ChangeFailureRateEnabled() {
+		buf.WriteByte('C')
+	}
+	if m.DevelopmentFrequencyEnabled() {
+		buf.WriteByte('D')
+	}
+	if m.ApplicationCountEnabled() {
+		buf.WriteByte('A')
+	}
+	return buf.String()
+}
+
+// Set parses the flag characters in m and updates *m.
+func (m *CollectorMode) Set(s string) {
```

Comment:
Instead of setting by a string (it will be hard for caller to remember what is `C`, `D`...), I think we can have something like this

``` golang
const (
   DeploymentFrequencyMetrics CollectorMetrics = 1 << iota
   ...
)

func (m CollectorMetrics) Enable(a CollectorMetrics) CollectorMetrics {
     return m | a
}

func (m CollectorMetrics) IsEnabled(a CollectorMetrics) bool {
   return m & a != 0
}
```

This is just my thought, please let me know your thoughts.
Thank you.

Reply:
Looks simpler and goood.